src/components/ShareStructure/ShareStructures.vue

```diff
@@ -0,0 +1,73 @@
+<template>
+  <div id="share-structures">
+
+    <resolution-date-dialog
+      attach="#share-structures"
+      :dialog="toggleResolutionDateDialog"
+      @emitClose="toggleResolutionDateDialog = false"
+    />
+
+    <share-structure
+      :is-edit-mode="isEditMode"
+      :isCorrection="isCorrectionView()"
+      :incorporation-application="getOriginalIA"
+      :share-classes="getShareClasses"
+      :business-snapshot="getOriginalSnapshot"
+      :resolution-required="resolutionsRequired"
+      :edit-label="editLabel"
+      :edited-label="editedLabel"
+      @emitShareClasses="setShareClasses($event)"
+      @emitShareStructureChanged="setShareStructureChanged($event)"
+      @emitEditingShareStructure="setEditingShareStructure($event)"
+      @emitResolutionPrompt="toggleResolutionDateDialog = $event"
+    />
+  </div>
+</template>
+<script lang="ts">
+// Libraries
+import { Component, Mixins, Prop } from 'vue-property-decorator'
+import { Action, Getter } from 'vuex-class'
+
+// Components
+import { ShareStructure } from '@bcrs-shared-components/share-structure'
+import { ResolutionDateDialog } from '@/components/dialogs'
+import { CommonMixin } from '@/mixins'
+
+import {
+  ActionBindingIF,
+  BusinessSnapshotIF,
+  IncorporationFilingIF,
+  ShareClassIF
+} from '@/interfaces'
+
+@Component({
+  components: {
+    ShareStructure,
+    ResolutionDateDialog
+  }
+})
+export default class ShareStructures extends Mixins(CommonMixin) {
+  /** Edit Mode */
+  @Prop({ default: true })
+  private isEditMode!: string
+
+  // Global Getters
+  @Getter getNewResolutionDates!: string []
+  @Getter getOriginalIA!: IncorporationFilingIF
+  @Getter getShareClasses!: ShareClassIF[]
+  @Getter getOriginalSnapshot!: BusinessSnapshotIF[]
+
+  // Global Actions
+  @Action setShareClasses!: ActionBindingIF
+  @Action setShareStructureChanged!: ActionBindingIF
+  @Action setEditingShareStructure!: ActionBindingIF
+
+  // Local Properties
+  private toggleResolutionDateDialog = false
+
+  /** Return true if changes to share structure rights will require a resolution date. */
+  get resolutionsRequired (): boolean {
+    return this.getNewResolutionDates.length === 0 && this.isAlterationView()
```

Comment:
Is there any chance `this.getNewResolutionDates` isn't an array? Is it initialized as `[]`?

Reply:
I don't believe so. It is initialized right in the store as `[]`. We only every add/remove the store property.

Reply:
I didn't see the new resolution dates in the store (but I didn't look very hard).